guillotina/api/suggestion.py

```diff
@@ -0,0 +1,58 @@
+from collections import Counter
+from guillotina import configure
+from guillotina.component import query_utility
+from guillotina.interfaces import ICatalogUtility
+from guillotina.interfaces import IResource
+
+
+@configure.service(
+    context=IResource,
+    method="GET",
+    permission="guillotina.AccessContent",
+    name="@suggestion",
+    summary="Make suggestion request",
+    responses={
+        "200": {
+            "description": "Suggestion results",
+            "content": {
+                "application/json": {
+                    "schema": {
+                        "type": "object",
+                        "properties": {
+                            "count": {"type": "integer"},
+                            "items": {"type": "array"}
+                        }
+                    }
+                }
+            }
+        }
+    },
+)
+async def suggestion_get(context, request):
+    query = request.query.copy()
+    search = query_utility(ICatalogUtility)
+    if search is None:
+        return {}
```

Comment:
I think it's confusing to return empty result when ICatalogUtility is None. You don't know if it's because the query doesn't match any result or you forgot to configure the catalog. I'd say the same for GET/POST @search endpoints.

What do you think?

Reply:
Agree